src/migratus/database.clj

```diff
@@ -114,15 +124,22 @@
         (find-init-script-file migration-dir init-script-name)
         (find-init-script-resource dir migration-dir init-script-name)))))
 
-(defn connect* [db]
+(defn connect* 
+  "Connects to the store - SQL database in this case.
+   Accepts a ^java.sql.Connection, ^javax.sql.DataSource or a db spec."
+  [db]
   (let [^Connection conn
         (cond
           (instance? Connection db) db
-          (instance? DataSource db) (try (.getConnection ^DataSource db)
-                                         (catch Exception e
-                                           (log/error e (str "Error getting DB connection from source" db))))
+          ;; TODO: ieugen: next.jdbc can get connection from DataSource, so we can drop this case perhaps
+          ;; https://cljdoc.org/d/com.github.seancorfield/next.jdbc/1.2.780/api/next.jdbc#get-connection
+          ;; (instance? DataSource db) (try (.getConnection ^DataSource db)
+          ;;                                (catch Exception e
+          ;;                                  (log/error e (str "Error getting DB connection from source" db))))
           :else (try
-                  (sql/get-connection db)
+                  ;; @ieugen: We can set auto-commit here as next.jdbc supports it. 
+                  ;; But I guess we need to conside the case when we get a Connection directly
```

Comment:
We could normalize the connection in the `let` above.

Reply:
Not sure right now but I feel like this is an improvement that can be done after the merge if needed.